dds/idl/be_produce.cpp

```diff
@@ -275,12 +275,49 @@ void postprocess(const char* fn, ostringstream& content,
 
 } // namespace
 
+inline bool
+contains_dcps_data_type(const char* line)
```

Comment:
I'd like to avoid having to parse this.  Could we change it to warn based on finding that there was a `#pragma` during the normal codegen passes (a visitor or generator class)?